specs/cli/client-add.md

```diff
@@ -0,0 +1,169 @@
+# kiota client add
+
+## Description 
+
+`kiota client add` allows a developer to add a new API client to the `kiota-config.json` file. If no `kiota-config.json` file is found, a new `kiota-config.json` file would be created. The command will add a new entry to the `clients` section of the `kiota-config.json` file. Once this is done, a local copy of the OpenAPI description is generated and kepts in the .
+
+When executing, a new API entry will be added and will use the `--client-name` parameter as the key for the map. When loading the OpenAPI description, it will generate a hash of the description to enable change detection of the description and save it as part of the `descriptionHash` property. It will also store the location of the description in the `descriptionLocation` property. If `--include-path` or `--exclude-path` are provided, they will be stored in the `includePatterns` and `excludePatterns` properties respectively.
```

Comment:
It also needs to support just a location. We'll add a `--search-key` parameter to record it in cases where it's coming from search. I would leave out the description source as I don't think it's required, knowing the search key contains the provider identifier. We would be able to report on "we couldn't find this search provider" when generating or refreshing later.

Reply:
if we don't include both the source and the key, it might break if a key changes.
e.g. we decide to publish descriptions for all national clouds, and in that process the key for Microsoft Graph (public) changes from microsoft-graph:v1 to microsoft-graph-public:v1. But we could rule that as an edge case and always add it later if we get feedback in that sense.

Reply:
Let's go with that for now.